src/GafferSceneUI/SceneView.cpp

```diff
@@ -1335,77 +1335,21 @@ void SceneView::frame( const GafferScene::PathMatcher &filter, const Imath::V3f
 
 void SceneView::expandSelection( size_t depth )
 {
-	Context::Scope scopedContext( getContext() );
-
 	PathMatcher &selection = const_cast<GafferScene::PathMatcherData *>( m_sceneGadget->getSelection() )->writable();
-	PathMatcher &expanded = expandedPaths()->writable();
 
-	std::vector<string> toExpand;
-	selection.paths( toExpand );
+	PathMatcher newlyExpandedLeaves = ContextAlgo::expandDescendants( getContext(), selection, preprocessedInPlug<ScenePlug>(), depth - 1 );
 
+	selection.clear();
 	bool needUpdate = false;
-	for( std::vector<string>::const_iterator it = toExpand.begin(), eIt = toExpand.end(); it != eIt; ++it )
+	for( PathMatcher::Iterator it = newlyExpandedLeaves.begin(), eIt = newlyExpandedLeaves.end(); it != eIt; ++it )
```

Comment:
Cleaner code (still clearing selection) in 75460fbf8f585b71677d2f1b10b1109de25969cb